src/types/log.rs

```diff
@@ -91,13 +114,21 @@ impl FilterBuilder {
 
     /// Single address
     pub fn address(mut self, address: Vec<H160>) -> Self {
-        self.filter.address = Some(address);
+        self.filter.address = Some(ValueOrArray(address));
         self
     }
 
     /// Topics
     pub fn topics(mut self, topic1: Option<Vec<H256>>, topic2: Option<Vec<H256>>, topic3: Option<Vec<H256>>, topic4: Option<Vec<H256>>) -> Self {
-        self.filter.topics = Some(vec![topic1, topic2, topic3, topic4]);
+        let mut topics = vec![topic1, topic2, topic3, topic4]
```

Comment:
Somewhat of a compromise to reduce breakage in the interface here, trims off the trailing `None` values from the topics as a shorter topic array will match regardless.

Think ideally this would be e.g. `pub fn topics(mut self, topcs: Vec<Option<Vec<H256>>>) -> Self` with perhaps `build()` checking that `topics.len() <= 4`, thoughts?

Reply:
Makes sense as well. I hope that Rust will allow us to omit `None` soon in the future by introducing optional parameters, though :)